lib/tx_build/soroban_authorization_entry.ex

```diff
@@ -65,4 +197,49 @@ defmodule Stellar.TxBuild.SorobanAuthorizationEntry do
     do: {:ok, root_invocation}
 
   defp validate_root_invocation(_root_invocation), do: {:error, :invalid_root_invocation}
+
+  @spec build_signature_args_from_xdr(
+          nonce :: Int64.t(),
+          signature_expiration_ledger :: UInt32.t(),
+          root_invocation :: SorobanAuthorizedInvocationXDR.t(),
+          secret_key :: secret_key()
+        ) :: SCVal.t() | error()
+  defp build_signature_args_from_xdr(
+         nonce,
+         signature_expiration_ledger,
+         root_invocation,
+         secret_key
+       ) do
+    {public_key, _secret_key} = KeyPair.from_secret_seed(secret_key)
+    raw_public_key = KeyPair.raw_public_key(public_key)
+    network_id = network_id_xdr()
+    envelope_type = EnvelopeType.new(:ENVELOPE_TYPE_SOROBAN_AUTHORIZATION)
+
+    signature =
+      network_id
+      |> Hash.new()
+      |> HashIDPreimageSorobanAuthorizationXDR.new(
+        nonce,
+        signature_expiration_ledger,
+        root_invocation
+      )
+      |> HashIDPreimageXDR.new(envelope_type)
+      |> HashIDPreimageXDR.encode_xdr!()
+      |> hash()
+      |> KeyPair.sign(secret_key)
```

Comment:
We should start the pipeline calling the `network_id_xdr()` function and avoid creating a new variable